### PR TITLE
Teamcapes plugin: Teamcape image instead of text

### LINKED ISSUE #5074: QoL PK'ing & Clanning Features Request
**Is your feature request related to a problem? Please describe.**
Freeze timers are incorrect, pretty frustrating. Teleblock timer is also inaccurate as well.

**Describe the solution you'd like**
- A simpler team-cape overlay, something that isn't so intrusive/large. The icon of the team-cape, for example team-37 cape. Rather than displaying team-37 cape: 1, have a small image of that team-cape with the number of that cape, followed by the quantity of people wearing that team-cape. Konduit does this very well.

- Menu-entry swap for clan-mates (without wearing the team-cape, so long as they're in the same clan chat) & friends, so avoid miss-clicking your clan-mates or friends if you're not wearing team-capes. https://i.imgur.com/OjrqMYJ.png - here is an example.

- Fixed freeze timers. The current barrage/freeze timers are always off by a few seconds. Also, a freeze timer for your opponent & when they're unfrozen would be a great feature.

- 
**Describe alternatives you've considered**
Other clients have these features.

**Additional context**
- Team-cape overlay re-design: https://i.imgur.com/OjrqMYJ.png
- Freeze timer displayed next to the opponent, freeze timers/tb timers fixed.

### PR DESCRIPTION
**Pull request by ItsSebas**
Makes teamcape overlay a bit more fancier.

Before:
image
After:
image
Partly closes #5074 
Thanks to FluffyRabbit and deathbeam for the help!
(LayoutTextComponent made by FluffyRabbit)

### CODE REVIEW

**Comment by deathbeam** [comment]
Even though this looks cool i think it should keep it's background at least for the parent panel

**Comment by raiyni** [comment]
I would set a limit on the number of capes displayed horizontally. Otherwise it will just keep stretching passed screen width.

**Comment by Nightfirecat** [comment]
I'd much prefer to see the numbers overlaid over the cape images, like you'd expect to see in a stackable item, or other item overlays like the jewellery charge overlay.

**Comment by ItsSebas** [comment]
The max amount of capes shown is set to 5 in other parts of the code.
I'll make numbers overlay like the slayer task plugin
image

**Comment by ItsSebas** [comment]
Panel now looks like this:
image

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 1]
```diff
@@ -26,19 +26,32 @@
...
 import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
 
 public class TeamCapesOverlay extends Overlay
 {
 	private final TeamCapesPlugin plugin;
 	private final TeamCapesConfig config;
-	private final PanelComponent panelComponent = new PanelComponent();
+	private PanelComponent verticalContainer = new PanelComponent();
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: don't need a vertical parent if you aren't displaying more than 1 row

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 2]
```diff
@@ -47,30 +60,33 @@
...
 		{
 			return null;
 		}
-		panelComponent.getChildren().clear();
-
 		for (Map.Entry<Integer, Integer> team : teams.entrySet())
 		{
-			// Only display team capes that have a count greater than the configured minimum.
-			if (team.getValue() >= config.getMinimumCapeCount())
+			// Only display team capes that have a count greater than the configured minimum AND teamcapes should be relevant to pking ( < 51)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: Just display all team capes and make the panel component to wrap after like 6 entries or something (it has property for it).

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java (hunk 1 above)
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: Make this final or even remove it yeah

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 3]
```diff
@@ -26,19 +26,32 @@
...
+import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
 
 public class TeamCapesOverlay extends Overlay
 {
 	private final TeamCapesPlugin plugin;
 	private final TeamCapesConfig config;
-	private final PanelComponent panelComponent = new PanelComponent();
+	private PanelComponent verticalContainer = new PanelComponent();
+	private PanelComponent horizontalContainer = new PanelComponent();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: Move this below injects and add empty line above it, also make it final and rename it t just panel

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 4]
```diff
@@ -26,19 +26,28 @@
...
 import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.OverlayManager;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
 
 public class TeamCapesOverlay extends Overlay
 {
-	private final TeamCapesPlugin plugin;
-	private final TeamCapesConfig config;
-	private final PanelComponent panelComponent = new PanelComponent();
+	@Inject
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: Move these new injections to constrcutor. and make ctor private

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 5]
```diff
@@ -47,30 +56,37 @@
 		setPriority(OverlayPriority.LOW);
 		this.plugin = plugin;
 		this.config = config;
+		panelComponent.setOrientation(PanelComponent.Orientation.HORIZONTAL);
+		panelComponent.setWrapping(4);
 	}
 
+	private final PanelComponent panelComponent = new PanelComponent();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: For moving stuff below injected fields i meant below fields not also below constructor......

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 6]
```diff
@@ -47,30 +56,37 @@
...
 			return null;
 		}
-		panelComponent.getChildren().clear();
-
 		for (Map.Entry<Integer, Integer> team : teams.entrySet())
 		{
-			// Only display team capes that have a count greater than the configured minimum.
-			if (team.getValue() >= config.getMinimumCapeCount())
+			// Only display team capes that have a count greater than the configured minimum AND teamcapes should be relevant to pking ( < 51)
+			if (team.getValue() >= config.getMinimumCapeCount() && team.getKey() < 51)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: This still contains the team.getKey check remove it

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 7]
```diff
@@ -47,30 +56,37 @@
...
-					.build());
+				//ItemID.TEAM1_CAPE == 4315, ItemID.TEAM2_CAPE == 4317, ItemID.TEAM50_CAPE == 4413
+				panelComponent.getChildren().add(new ImageComponent(getImage(( team.getKey() * 2 ) + 4313 ,  team.getValue())));
 			}
 		}
-
 		return panelComponent.render(graphics);
 	}
+
+	private BufferedImage getImage(int itemID, int amount)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: THis method is like super unnecessary

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 8]
```diff
@@ -28,49 +28,54 @@
...
 import net.runelite.client.ui.overlay.Overlay;
 import net.runelite.client.ui.overlay.OverlayPosition;
 import net.runelite.client.ui.overlay.OverlayPriority;
-import net.runelite.client.ui.overlay.components.LineComponent;
+import net.runelite.client.ui.overlay.components.ImageComponent;
 import net.runelite.client.ui.overlay.components.PanelComponent;
 
 public class TeamCapesOverlay extends Overlay
 {
+
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: Remove this empty line

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 9]
```diff
@@ -28,49 +28,53 @@
...
 		this.config = config;
+		this.manager = manager;
+		panelComponent.setOrientation(PanelComponent.Orientation.HORIZONTAL);
+		panelComponent.setWrapping(4);
 	}
 
 	@Override
 	public Dimension render(Graphics2D graphics)
 	{
+		panelComponent.getChildren().clear();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: Any reason why this was moved above the check?

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 10]
```diff
@@ -28,49 +28,53 @@
...
 		{
-			// Only display team capes that have a count greater than the configured minimum.
+			// Only display team capes that have a count greater than the configured minimum
 			if (team.getValue() >= config.getMinimumCapeCount())
 			{
-				panelComponent.getChildren().add(LineComponent.builder()
-					.left("Team-" + Integer.toString(team.getKey()))
-					.right(Integer.toString(team.getValue()))
-					.build());
+				//ItemID.TEAM1_CAPE == 4315 (1), ItemID.TEAM2_CAPE == 4317 (2), ItemID.TEAM50_CAPE == 4413 (50)
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: what are these comments? remove them

File: runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java [hunk 11]
```diff
@@ -28,49 +28,53 @@
...
+				//ItemID.TEAM_CAPE_ZERO == 20211 (51), ItemID.TEAM_CAPE_X == 20214 (52), ItemID.TEAM_CAPE_I == 20217 (53)
+				int teamcapeNumber = team.getKey();
+				int ItemID = teamcapeNumber < 51 ? (2 * teamcapeNumber + 4313) : (3 * (teamcapeNumber - 50) + 20208);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teamcapes/TeamCapesOverlay.java]: itemId not ItemId

**Pull request merged**